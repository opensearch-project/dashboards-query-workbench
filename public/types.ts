/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { DataSourcePluginStart } from '../../../src/plugins/data_source/public/types';
import { NavigationPublicPluginStart } from '../../../src/plugins/navigation/public';
import { ObservabilityStart } from './types';

export interface WorkbenchPluginSetup {}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface WorkbenchPluginStart {
}

export interface AppPluginStartDependencies {
  navigation: NavigationPublicPluginStart;
  dataSource: DataSourcePluginStart;
  observabilityDashboards?: ObservabilityStart;
}

export type { ObservabilityStart } from '../../dashboards-observability/public';
