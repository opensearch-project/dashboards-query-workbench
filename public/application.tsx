/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */


import React from 'react';
import ReactDOM from 'react-dom';
import { AppMountParameters, CoreStart } from '../../../src/core/public';
import { DataSourceManagementPluginSetup } from '../../../src/plugins/data_source_management/public';
import { WorkbenchApp } from './components/app';
import { AppPluginStartDependencies } from './types';

export const renderApp = (
  { notifications, http, chrome, savedObjects }: CoreStart,
  { navigation, dataSource }: AppPluginStartDependencies,
  { appBasePath, element, setHeaderActionMenu }: AppMountParameters,
  dataSourceManagement: DataSourceManagementPluginSetup
) => {
  ReactDOM.render(
    <WorkbenchApp
      basename={appBasePath}
      notifications={notifications}
      http={http}
      navigation={navigation}
      chrome={chrome}
      savedObjects={savedObjects}
      dataSourceEnabled={!!dataSource}
      dataSourceManagement={dataSourceManagement}
      setActionMenu={setHeaderActionMenu}
    />,
    element
  );

  return () => ReactDOM.unmountComponentAtNode(element);
};
