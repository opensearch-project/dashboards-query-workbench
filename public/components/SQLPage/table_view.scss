/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

.workbench-tree {
  .euiTreeView__node--expanded {
    max-height: none !important; // Fixes OUI issue: https://github.com/opensearch-project/oui/issues/1118
  }
}
