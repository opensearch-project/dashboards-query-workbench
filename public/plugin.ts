/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { AppMountParameters, CoreSetup, CoreStart, Plugin } from '../../../src/core/public';
import { DataSourcePluginSetup, DataSourcePluginStart } from '../../../src/plugins/data_source/public';
import { DataSourceManagementPluginSetup } from '../../../src/plugins/data_source_management/public';
import { PLUGIN_NAME } from '../common/constants';
import { registerObservabilityDependencies } from './dependencies/register_observability_dependencies';
import { coreRefs } from './framework/core_refs';
import { AppPluginStartDependencies, WorkbenchPluginSetup, WorkbenchPluginStart } from './types';
export interface WorkbenchPluginSetupDependencies {
  dataSource: DataSourcePluginSetup;
  dataSourceManagement: DataSourceManagementPluginSetup
}

export interface WorkbenchPluginStartDependencies {
  dataSource: DataSourcePluginStart;
}
export class WorkbenchPlugin implements Plugin<WorkbenchPluginSetup, WorkbenchPluginStart> {
  public setup(core: CoreSetup, {dataSource, dataSourceManagement} : WorkbenchPluginSetupDependencies): WorkbenchPluginSetup {
    // Register an application into the side navigation menu
    core.application.register({
      id: 'opensearch-query-workbench',
      title: PLUGIN_NAME,
      category: {
        id: 'opensearch',
        label: 'OpenSearch Plugins',
        order: 2000,
      },
      order: 1000,
      async mount(params: AppMountParameters) {
        // Load application bundle
        const { renderApp } = await import('./application');
        // Get start services as specified in opensearch_dashboards.json
        const [coreStart, depsStart] = await core.getStartServices();
        // Render the application
        return renderApp(coreStart, depsStart as AppPluginStartDependencies, params, dataSourceManagement);
      },
    });

    // Return methods that should be available to other plugins
    return {};
  }

  public start(core: CoreStart, startDeps: AppPluginStartDependencies): WorkbenchPluginStart {
    coreRefs.http = core.http;
    coreRefs.savedObjectsClient = core.savedObjects.client;
    coreRefs.toasts = core.notifications.toasts;
    coreRefs.chrome = core.chrome;
    coreRefs.application = core.application;
    coreRefs.overlays = core.overlays;
    coreRefs.dataSource = startDeps.dataSource;

    registerObservabilityDependencies(startDeps.observabilityDashboards);
    return {};
  }

  public stop() {}
}
