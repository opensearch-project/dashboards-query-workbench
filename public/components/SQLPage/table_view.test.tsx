import '@testing-library/jest-dom';
import React from 'react';
import { httpClientMock } from '../../../test/mocks';

import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { HttpResponse } from '../../../../../src/core/public';
import { mockDatabaseQuery, mockJobId, mockOpenSearchIndicies } from '../../../test/mocks/mockData';
import { TableView } from './table_view';

describe('Render databases in tree', () => {
  it('fetches and displays indicies when datasource is OpenSearch', async () => {
    const client = httpClientMock;
    client.post = jest.fn(() => {
      return (Promise.resolve(mockOpenSearchIndicies) as unknown) as HttpResponse;
    });

    const asyncTest = () => {
      render(
        <TableView
          http={client}
          selectedItems={[{ label: 'OpenSearch' }]}
          updateSQLQueries={() => {}}
          refreshTree={false}
        />
      );
    };
    await asyncTest();
    expect(document.body.children[0]).toMatchSnapshot();
  });
  it('fetches and displays database nodes when datasource is s3', async () => {
    const client = httpClientMock;
    client.post = jest.fn(() => {
      return (Promise.resolve(mockJobId) as unknown) as HttpResponse;
    });
    client.get = jest.fn(() => {
      return (Promise.resolve(mockDatabaseQuery) as unknown) as HttpResponse;
    });

    const asyncTest = () => {
      render(
        <TableView
          http={client}
          selectedItems={[{ label: 'my_glue' }]}
          updateSQLQueries={() => {}}
          refreshTree={false}
        />
      );
    };
    await asyncTest();
    expect(document.body.children[0]).toMatchSnapshot();
  });
});
