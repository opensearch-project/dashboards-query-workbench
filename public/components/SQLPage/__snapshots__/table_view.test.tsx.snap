// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render databases in tree fetches and displays database nodes when datasource is s3 1`] = `
<div>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiEmptyPrompt"
      >
        <svg
          aria-hidden="true"
          class="euiIcon euiIcon--xxLarge euiIcon--danger"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.59 10.059 7.35 5.18h1.3L8.4 10.06h-.81Zm.394 1.901a.61.61 0 0 1-.448-.186.606.606 0 0 1-.186-.444c0-.174.062-.323.186-.446a.614.614 0 0 1 .448-.184c.169 0 .315.06.44.182.124.122.186.27.186.448a.6.6 0 0 1-.189.446.607.607 0 0 1-.437.184ZM2 14a1 1 0 0 1-.878-1.479l6-11a1 1 0 0 1 1.756 0l6 11A1 1 0 0 1 14 14H2Zm0-1h12L8 2 2 13Z"
            fill-rule="evenodd"
          />
        </svg>
        <div
          class="euiSpacer euiSpacer--m"
        />
        <h3
          class="euiTitle euiTitle--medium"
        >
          Error loading data
        </h3>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render databases in tree fetches and displays indicies when datasource is OpenSearch 1`] = `
<div>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiText euiText--medium euiTreeView__wrapper"
      >
        <p
          class="euiScreenReaderOnly"
          id="some_html_id--instruction"
        >
          You can quickly navigate this list using arrow keys.
        </p>
        <ul
          aria-describedby="some_html_id--instruction"
          aria-label="Sample Folder Tree"
          class="euiTreeView"
          id="some_html_id"
        >
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_0"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .kibana_1
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_1"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .kibana_2
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_2"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .kibana_3
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_3"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .opendistro-reports-definitions
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_4"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .opendistro-reports-instances
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_5"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .opensearch-observability
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_6"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .plugins-ml-config
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_7"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .ql-datasources
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_8"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .ql-job-metadata
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_9"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .query_execution_result
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_10"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      flint_mys3_default_alb_logs_temp_10_test_index
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_11"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      flint_mys3_default_alb_logs_temp_5_mys3_default...
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_12"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      flint_mys3_default_elb_logs_regex_elb_logs_rege...
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_13"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      flint_mys3_default_empty_table_empty_table_inde...
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_14"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      flint_mys3_default_http_logs_skipping_index
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_15"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      flint_mys3_default_parquet_elb_logs_simple_parq...
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_16"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      opensearch_dashboards_sample_data_flights
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_17"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      opensearch_dashboards_sample_data_logs
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_18"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      ss4o_traces-elb-test
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_19"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .flint-elv-mv
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
          <li
            class="euiTreeView__node"
          >
            <button
              aria-controls="some_html_id"
              aria-expanded="false"
              class="euiTreeView__nodeInner"
              data-test-subj="euiTreeViewButton-some_html_id"
              id="element_20"
            >
              <span
                class="euiTreeView__iconWrapper"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiIcon-isLoading"
                  focusable="false"
                  height="16"
                  role="img"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                />
              </span>
              <span
                class="euiTreeView__nodeLabel"
              >
                <div>
                  <span
                    class="euiToolTipAnchor"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      .kibana
                    </div>
                  </span>
                   
                </div>
              </span>
            </button>
            <div
              id="some_html_id"
            />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
